src/mono/wasm/runtime/jiterpreter.ts

```diff
@@ -2925,6 +2949,13 @@ function append_bailout (builder: WasmBuilder, ip: MintOpcodePtr, reason: Bailou
     builder.appendU8(WasmOpcode.return_);
 }
 
+function append_safepoint (builder: WasmBuilder, ip: MintOpcodePtr) {
+    builder.local("frame");
+    // Not ip_const, because we can't pass relative IP to do_safepoint
+    builder.i32_const(ip);
+    builder.callImport("safepoint");
```

Comment:
Would be nice to inline the if (G_UNLIKELY (...)) part of the safepoint code here, its almost always false.

Reply:
Is there some way for us to export the flag as a wasm global? If so, I could grab it out of the module and import it.

Reply:
It has an address in linear memory, so just taking that address and loading from it should work.

Reply:
Good point, that's probably only slightly slower, and I can just hand the address out. Thanks.